syntax = "proto3";

package ptre;

service Tcp {
  rpc PullTensor (PullTensorRequest) returns (PullTensorResponse) {}
  rpc PushTensor(PushTensorRequest) returns (PushTensorResponse) {}
}

message PullTensorRequest{
  int32 src_rank = 1;
  string tensor_name = 2;
}

message PullTensorResponse{
  int32 dst_rank = 1;
  string tensor_name = 2;
  bytes buf = 3;
  int32 status = 4;
}

message PushTensorRequest {
  int32 src_rank = 1;
  string tensor_name = 2;
  bytes buf = 3;
}

message PushTensorResponse {
  int32 dst_rank = 1;
  string tensor_name = 2;
  int32 status = 3;
}
